lib/libvfio-user.c

```diff
@@ -661,31 +654,13 @@ consume_fd(int *fds, size_t nr_fds, size_t index)
    return fd;
 }
 
-int
-handle_dma_map(vfu_ctx_t *vfu_ctx, vfu_msg_t *msg,
-               struct vfio_user_dma_map *dma_map)
+static int
+handle_dma_map_quiesced(vfu_ctx_t *vfu_ctx, vfu_msg_t *msg,
+                        struct vfio_user_dma_map *dma_map)
 {
-    char rstr[1024];
-    int fd = -1;
-    int ret;
+    int ret, idx, fd = -1;
     uint32_t prot = 0;
 
-    assert(vfu_ctx != NULL);
-    assert(msg != NULL);
-    assert(dma_map != NULL);
-
-    if (msg->in_size < sizeof(*dma_map) || dma_map->argsz < sizeof(*dma_map)) {
-        vfu_log(vfu_ctx, LOG_ERR, "bad DMA map region size=%zu argsz=%u",
-                msg->in_size, dma_map->argsz);
-        return ERROR_INT(EINVAL);
-    }
-
-    snprintf(rstr, sizeof(rstr), "[%#lx, %#lx) offset=%#lx flags=%#x",
```

Comment:
I think it's better to have this output at the point we're about to handle it, not the point that requested it. (see comment about quiesce below too)

Reply:
OK, moved right before we call `dma_controller_add_region`.